var/spack/repos/builtin/packages/py-h5py/package.py

```diff
@@ -44,12 +50,26 @@ class PyH5py(PythonPackage):
     depends_on('mpi', when='+mpi')
     depends_on('py-mpi4py', when='+mpi', type=('build', 'run'))
 
+    # Conflicts
+    conflicts('^hdf5@1.12.0:', when='@:2.999.999')
+
     phases = ['configure', 'install']
 
+    @when('@:2.999.999')
     def configure(self, spec, prefix):
         self.setup_py('configure', '--hdf5={0}'.format(spec['hdf5'].prefix),
                       '--hdf5-version={0}'.format(spec['hdf5'].version))
 
         if '+mpi' in spec:
             env['CC'] = spec['mpi'].mpicc
             self.setup_py('configure', '--mpi')
+
+    @when('@3.0.0:')
+    def configure(self, spec, prefix):
+        if '+mpi' in spec:
+            env['CC'] = spec['mpi'].mpicc
+            env['HDF5_MPI'] = 'ON'
+            env['HDF5_DIR'] = spec['hdf5'].prefix
+
+    def install(self, spec, prefix):
+        self.setup_py("install", "--prefix={0}".format(prefix))
```

Comment:
Why is the new `install` not needed for older versions?

Reply:
The `configure` step was removed the configuration now occurs via environment variables. It was easier to have the two phase install procedure as is `[configure, install]` instead of changing the number of phases depending upon the version. 

https://github.com/h5py/h5py/blob/master/docs/whatsnew/3.0.rst#building-h5py

Reply:
This is basically the default `install` method, can we remove this?

Reply:
Done.